packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -626,6 +626,62 @@ class SliverChildListDelegate extends SliverChildDelegate {
   final SemanticIndexCallback semanticIndexCallback;
 
   /// The widgets to display.
+  ///
+  /// If this list is going to be mutated, it is usually wise to put a [Key] on
+  /// each of the child widgets, so that the framework can match old
+  /// configurations to new configurations and maintain the underlying render
+  /// objects.
+  ///
+  /// Also, a [Widget] in Flutter is immutable, so directly modifying the
+  /// [children] such as `someWidget.children.add(...)` or
+  /// as the example code below will result in incorrect behaviors. Whenever the
+  /// children list is modified, a new list object should be provided.
+  ///
+  /// ```dart
+  /// class SomeWidgetState extends State<SomeWidget> {
+  ///   List<Widget> _children;
+  ///
+  ///   void initState() {
+  ///     _children = [];
+  ///   }
+  ///
+  ///   void someHandler() {
+  ///     setState(() {
+  ///         _children.add(ChildWidget());
+  ///     });
+  ///   }
+  ///
+  ///   Widget build(BuildContext context) {
+  ///     // Reusing `List<Widget> _children` here is problematic.
+  ///     return PageView(children: _children);
+  ///   }
+  /// }
+  /// ```
```

Comment:
I think we should just drop the whole "bad" block - some people might miss that it's bad and try to use it or base their code off of it. Instead we should add comments below about why we're using a key, and why we're creating a new list object in build.